src/Panel/Ui/FilePreview.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Kirby\Panel\Ui;
+
+use Kirby\Cms\App;
+use Kirby\Cms\File;
+use Kirby\Exception\InvalidArgumentException;
+use Kirby\Exception\LogicException;
+use Kirby\Toolkit\I18n;
+
+/**
+ * Defines a component that implements a file preview
+ *
+ * @package   Kirby Panel
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ * @since     5.0.0
+ * @internal
+ */
+class FilePreview extends Component
+{
+	public function __construct(
+		public File $file,
+		public string $component = 'k-file-default-preview'
+	) {
+	}
+
+	/**
+	 * Returns true if this class should
+	 * handle the preview of this file
+	 *
+	 * @codeCoverageIgnore
+	 */
+	public static function accepts(File $file): bool
+	{
+		throw new LogicException('Filew preview classes must define the static ::accepts() method');
```

Comment:
I think this went wrong with the merge …
```suggestion
		return true;
```

Reply:
Mmm need to have a look why. I wanted to avoid your solution as any customer file preview not adding their own accepts method will claim any file. And they shouldn't. So i would really like to enforce them writing the accepts method (which is why I went abstract class in the first place).

Reply:
Ah, now I get it. That explains the abstract class quite well. I'm not too sure though that we can enforce it like that. I think the risk still remains that people would just extend some other existing FilePreview class or the default class instead and the same problem would happen there as well. It's probably more about documenting the extension process correctly.

Reply:
I agree on having the one class. What I didn't understand why you'd prefer to not throw an exception when no custom accepts method has been defined.

Reply:
Maybe I'm misunderstanding the concept, but wouldn't our default file preview class also need to define which types of files it can handle with the accept method?

Reply:
@bastianallgeier My ideas was different:
- `FilePreview::factory()` loops through all file preview extensions and calls their `::accepts()` method
- The first returning true gets chosen, a new object created and returned
- `FilePreview` itself isn't in that list of file preview extensions.
- Instead if the loop runs through without finding any match, we create an object with the base class and return it.

https://github.com/getkirby/kirby/pull/6575/files#diff-bfc2ed602ed817e9a56aa6e78e991c493524a5948405519d87d8d9c20fc224b7R77-R87

Reply:
I feel really bad about it, but after understanding it better, I think your initial setup with the abstract indeed made more sense 🙈 sorry for letting you jump through those hoops.

Reply:
Changed back 😅